examples/prediction/category_prediction/category.py

```diff
@@ -0,0 +1,188 @@
+
+This model uses the [MSNBC.com Anonymous Web Data Data Set][1] provided by
+[UCI Machine Learning Repository][2] to predict the next page the user is more
+likely to click. Each page is assigned a category and the user behavior is
+recorded as navigating from one page category to another. The dataset contains
+one user session per line. See [dataset][1] description for more information.
+
+For the purpose of this demonstration, the model will be trained on the
+first 100 user sessions and try to predict a random session from the learned
+sessions.
+
+References:
+  1: https://archive.ics.uci.edu/ml/datasets/MSNBC.com+Anonymous+Web+Data
+  2: Lichman, M. (2013). UCI Machine Learning Repository [http://archive.ics.uci.edu/ml].
+     Irvine, CA: University of California, School of Information and Computer Science
+"""
+import os
+import random
+import sys
+import zipfile
+from operator import itemgetter
+
+from nupic.frameworks.opf.model_factory import ModelFactory
+
+# List of categories in the dataset
+CATEGORIES = {
+  "1": "frontpage", "2": "news", "3": "tech", "4": "local", "5": "opinion",
+  "6": "on-air", "7": "misc", "8": "weather", "9": "msn-news", "10": "health",
+  "11": "living", "12": "business", "13": "msn-sports", "14": "sports",
+  "15": "summary", "16": "bbs", "17": "travel"
+}
+
+# HTM Prediction Model Parameters
+MODEL_PARAMS = {
+  "model": "HTMPrediction",
+  "version": 1,
+  "predictAheadTime": None,
+
+  "modelParams": {
+    "inferenceType": "TemporalMultiStep",
+    "sensorParams": {
+      "verbosity": 0,
+      "encoders": {
+
+        # Category encoder used for web page categories
+        "category": {
+          "fieldname": u"category",
+          "name": u"category",
+          "type": "CategoryEncoder",
+          "categoryList": CATEGORIES.keys(),
+          "w": 21
+        },
+      },
+      "sensorAutoReset": None,
+    },
+
+    # No need to use SP for this dataset
+    "spEnable": False,
+    "spParams": {
+    },
+    "tmEnable": True,
+    "tmParams": {
+      'seed': 1960,
+      'temporalImp': 'cpp',
+      'globalDecay': 0.0,
+      'initialPerm': 0.21,
+      'inputWidth': 2048,
+      'maxAge': 0,
+      'outputType': 'normal',
+      'pamLength': 2,
+      'permanenceDec': 0.1,
+      'permanenceInc': 0.1,
+      'verbosity': 0,
+      'activationThreshold': 13,
+      'cellsPerColumn': 32,
+      'columnCount': 2048,
+      'maxSegmentsPerCell': 128,
+      'maxSynapsesPerSegment': 32,
+      'minThreshold': 10,
+      'newSynapseCount': 20,
+
+    },
+    "clParams": {
+      'implementation': 'cpp',
+      "regionName": "SDRClassifierRegion",
+      "verbosity": 0,
+      "alpha": 0.01,
+      "steps": "1",
+    },
+  },
+}
+
+
+
+def main():
+  # Create HTM Prediction model and enable inference on the "category" field
+  model = ModelFactory.create(MODEL_PARAMS)
+  model.enableInference({"predictedField": "category"})
+
+  # Assumes data file is in same directory as the script
```

Comment:
I'm assuming the user has to download the dataset ahead of time. If the file is not present we should let them know in an error message.

Reply:
The datafile is in the repo. No need to download.